### PR TITLE
Added account name filter to SMS Billing Report

### PR DESCRIPTION
**Pull request by michelleliang1**
https://dimagi-dev.atlassian.net/browse/SAAS-11011

##### SUMMARY

An account name filter was added to shorten the time it takes to generate a SMS billing report. 
I used the NameFilter from accounting for similar use in smsbillables. 
  

##### FEATURE FLAG
N/A

##### RISK ASSESSMENT / QA PLAN

Requires QA before merge: https://dimagi-dev.atlassian.net/jira/software/projects/QA/boards/3/backlog

##### PRODUCT DESCRIPTION

Users are able to filter SMS billables by account name.

### CODE REVIEW

**Review by millerdev — COMMENTED** [review]
I don't feel like I have much context in this area, so will allow others to review. I'm assuming this area is untested? It would be nice to have tests, but understandable if they are not added because it's a hard area to test.

File: corehq/apps/smsbillables/interface.py [hunk 1]
```diff
@@ -129,6 +140,7 @@ def _format_billables(self, sms_billables):
         return [
             [
                 sms_billable.date_sent,
+                BillingAccount.get_account_by_domain(sms_billable.domain).name,
```
Comment (dannyroberts) [inline corehq/apps/smsbillables/interface.py:143]: I'm wondering if this might be prohibitively expensive (time-wise) to look up separately for each row. Since the domain-to-account mapping is fairly stable (i.e. it does not change frequently), it's a good candidate for caching, which should be straightforward to add. I think just decorating the function definition of `get_account_by_domain` with something like `@quickcache(['domain'])` would do it

Comment (michelleliang1) [reply corehq/apps/smsbillables/interface.py:143]: @dannyroberts Added!

Comment (dannyroberts) [reply corehq/apps/smsbillables/interface.py:143]: Looks like my suggestion didn't work as simply as I'd hoped. Apparently this is because `get_account_by_domain` is written to accept _either_ a domain name (which is what I thought and would work fine with quickcache) _or_ a domain object (which is sometimes used, and which doesn't play nicely with quickcache).

I think an easy workaround that still lets us cache here is to pull `BillingAccount.get_account_by_domain(sms_billable.domain).name` out into a function that takes a domain name (_name_ only, never object) and returns an account name. And then put `@quickcache(['domain'])` on _that_. Alternatively you can just revert the caching change altogether. Given that it's just one page of results at a time it has to do this for (right?) I may have been over-optimizing on what might not even be the slowest part of the workflow. Given that this is an internal UI, as long as we test it for reasonable speed (like it doesn't time out) after deploying this, that's fine.

**Comment by dannyroberts** [comment]
Looks like just one change needed to get tests passing, which I think was the only thing blocking here?

**Comment by michelleliang1** [comment]
Also working on fixing some QA finds!

**Comment by michelleliang1** [comment]
@dannyroberts Changes updated and ready for review!

File: corehq/apps/accounting/models.py [hunk 2]
```diff
@@ -459,7 +459,6 @@ def get_or_create_account_by_domain(cls, domain,
         ), True
 
     @classmethod
-    @quickcache(['domain'])
```
Comment (dannyroberts) [inline corehq/apps/accounting/models.py]: nit: these should probably be two separate commits (text color / revert caching). You can use `git add -p` to select which local changes you want to add to a commit.  That's mostly just a tip going forward; this PR doesn't need to block on cleaning up the commit history or anything

**Review by dannyroberts — APPROVED** [review]
